lib/client/ipam_test.go

```diff
@@ -103,6 +103,70 @@ type testArgsClaimAff struct {
 
 var _ = Describe("IPAM tests", func() {
 
+	Describe("IPAM AutoAssign from the default pool then delete the pool and assign again", func() {
+		testutils.CleanEtcd()
+		c, _ := testutils.NewClient("")
+		ic := setupIPAMClient(true)
+
+		host := "host-A"
+		pool1 := testutils.MustParseCIDR("10.0.0.0/24")
+
+		testutils.CreateNewIPPool(*c, "10.0.0.0/24", false, false, true)
+
+		// Step-1: AutoAssign 1 IP without specifying a pool - expect the assigned IP is from pool1.
+		Context("AutoAssign 1 IP without specifying a pool", func() {
+			args := client.AutoAssignArgs{
+				Num4:     1,
+				Num6:     0,
+				Hostname: host,
+			}
+
+			v4, _, outErr := ic.AutoAssign(args)
+
+			It("assigned IP should be from pool1", func() {
+				Expect(outErr).NotTo(HaveOccurred())
+				Expect(pool1.IPNet.Contains(v4[0].IP)).To(BeTrue())
+			})
+		})
+
+		// // Step-1.5: Have to create a node with name="host-A" so release affinity works.
+		// c.Nodes().Create(&api.Node{
+		// 	Metadata: api.NodeMetadata{Name: host},
+		// 	Spec:     api.NodeSpec{},
+		// })
+
+		// Step-2: Delete pool1 - expect it to execute without any error.
+		Context("Delete pool1", func() {
+			outErr := c.IPPools().Delete(api.IPPoolMetadata{
+				CIDR: pool1,
+			})
+
+			It("should delete the ipPool without any error", func() {
+				Expect(outErr).NotTo(HaveOccurred())
+			})
+		})
```

Comment:
Is there anything we can do to check that it has removed affinities associated with those pools?